app/logic/Mail/EWS/EWSAccount.ts

```diff
@@ -529,14 +522,55 @@ export class EWSAccount extends MailAccount {
         }
         ewsFolder.fromXML(folder);
         this.folderMap.set(folder.FolderId.Id, ewsFolder);
-      } // else TODO set calendar.folderID and addressbook.folderID
+      }
     }
     // Iterate from deepest to shallowest
     for (let folder of this.getAllFolders().reverse()) {
       if (!this.folderMap.has(folder.id)) {
         await folder.deleteItLocally();
       }
     }
+    for (let folder of ensureArray(result.RootFolder.Folders.ContactsFolder)) {
+      // Link (until #155) or create the default address book.
+      // TODO: Support user-added address books. Compare FolderId.
+      // FolderClass will be IPF.Contacts but some internal folders exist
+      if (folder.DistinguishedFolderId == "contacts") {
+        let addressbook = appGlobal.addressbooks.find(addressbook => addressbook.mainAccount == this) as EWSAddressbook | null;
+        if (!addressbook) {
+          addressbook = newAddressbookForProtocol("addressbook-ews") as EWSAddressbook;
+          addressbook.name = this.name;
+          addressbook.url = this.url;
+          addressbook.username = this.username;
+          addressbook.workspace = this.workspace;
+          addressbook.mainAccount = this;
+          appGlobal.addressbooks.add(addressbook);
+        }
+        addressbook.folderID = folder.FolderId.Id;
```

Comment:
Move into `if`. Should read from DB, where possible.
same elsewhere (2x2 places)

Reply:
Existing accounts won't have a `folderID` yet, and this will ensure that it gets set for now; future support for multiple contacts/calendars will fix this properly.